android/app/src/main/java/org/uwaterloo/subletr/pages/createlisting/CreateListingPageView.kt

```diff
@@ -0,0 +1,703 @@
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.ModalBottomSheet
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.SheetState
+import androidx.compose.material3.Text
+import androidx.compose.material3.rememberDateRangePickerState
+import androidx.compose.material3.rememberModalBottomSheetState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.rxjava3.subscribeAsState
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.asImageBitmap
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.LiveRegionMode
+import androidx.compose.ui.semantics.clearAndSetSemantics
+import androidx.compose.ui.semantics.liveRegion
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import kotlinx.coroutines.launch
+import org.uwaterloo.subletr.R
+import org.uwaterloo.subletr.components.button.PrimaryButton
+import org.uwaterloo.subletr.components.textfield.RoundedTextField
+import org.uwaterloo.subletr.enums.HousingType
+import org.uwaterloo.subletr.theme.SubletrTheme
+import org.uwaterloo.subletr.theme.secondaryTextColor
+import org.uwaterloo.subletr.theme.subletrPink
+import org.uwaterloo.subletr.theme.textFieldBackgroundColor
+import org.uwaterloo.subletr.theme.textFieldBorderColor
+import org.uwaterloo.subletr.theme.textOnSubletrPink
+import org.uwaterloo.subletr.utils.toBase64String
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun CreateListingPageView(
+	modifier: Modifier = Modifier,
+	viewModel: CreateListingPageViewModel = hiltViewModel(),
+	uiState: CreateListingPageUiState = viewModel.uiStateStream.subscribeAsState(
+		CreateListingPageUiState.Loading
+	).value,
+) {
+	val scrollState = rememberScrollState()
+	val coroutineScope = rememberCoroutineScope()
+
+	val dateRangePickerState = rememberDateRangePickerState()
+	var startButtonText by remember { mutableStateOf("") }
+	var endButtonText by remember { mutableStateOf("") }
```

Comment:
Sorry, on quick glance, I can't really tell why this isn't a stream. Could you explain?

Reply:
not sure what you mean, I guess those variables could be a stream but I'm not too sure about the difference in doing a stream vs. this method.
Also idk if this is what you are asking but I didn't put this as a stream in the ViewModel because these variables are only for frontend